dags/iop/parser.py

```diff
@@ -49,10 +54,24 @@ def _get_journal_doctype(self, article: ET.Element):
             self.logger.error("Article-type is not found in XML", doi=self.dois)
             return None
         try:
-            return self.article_type_mapping[value]
+            self.journal_doctype = self.article_type_mapping[value]
+            return self.journal_doctype
         except KeyError:
             self.logger.error(
                 "Unmapped article type", doi=self.dois, article_type=value
             )
         except Exception:
             self.logger.error("Unknown error", doi=self.dois)
+
+    def _get_related_article_doi(self, article):
+        if self.journal_doctype in ["corrigendum", "addendum"]:
+            node = article.find(
```

Comment:
actually, you can return if the `self.journal_doctype not in ["corrigendum", "addendum"]` and then do other stuff. Then no need for `return` cause in Python when you don't return you `None` by default